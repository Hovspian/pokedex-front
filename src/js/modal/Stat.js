import React from 'react';
import PropTypes from 'prop-types';

import '../../styles/modal/Stat.css'

const propTypes = {
  statAmount: PropTypes.number.isRequired,
};

class Stat extends React.Component {
  render() {
    return (
      <div className="stat" style={{ "height": `${this.props.statAmount}px` }}>
        <span className="amount">{this.props.statAmount}</span>
      </div>
    );
  }
}

Stat.propTypes = propTypes

export default Stat;
