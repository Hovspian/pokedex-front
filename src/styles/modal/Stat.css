.stat {
  width: 11%;
  background-color: rgb(3, 128, 253);
  color: white;
  align-self: flex-end;
  display: table;
  border-width: 1px 1px 0 1px;
  border-style: solid;
  border-color: black;
  max-height: 100%;
  min-height: 30px;
}

.amount {
  justify-content: center;
  display: table-cell;
  vertical-align: middle;
}
