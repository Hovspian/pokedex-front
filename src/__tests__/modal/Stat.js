import React from 'react';
import { shallow } from 'enzyme';

import Stat from '../../js/modal/Stat';

describe('Stat Component', () => {
  it('should render Stat component without crashing', () => {
    shallow(<Stat statAmount={10} />);
  });

  it('should match Stat component snapshot', () => {
    const wrapper = shallow(<Stat statAmount={10} />);
    expect(wrapper).toMatchSnapshot();
  });
});
