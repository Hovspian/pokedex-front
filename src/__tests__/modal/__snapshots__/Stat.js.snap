// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat Component should match Stat component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stat
    statAmount={10}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="amount"
>
        10
</span>,
      "className": "stat",
      "style": Object {
        "height": "10px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 10,
        "className": "amount",
      },
      "ref": null,
      "rendered": 10,
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="amount"
>
          10
</span>,
        "className": "stat",
        "style": Object {
          "height": "10px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 10,
          "className": "amount",
        },
        "ref": null,
        "rendered": 10,
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
